core/includes/update.inc

```diff
@@ -146,24 +146,34 @@ function update_prepare_bootstrap() {
       // process.
       backdrop_bootstrap(BACKDROP_BOOTSTRAP_VARIABLES);
 
-      // Change language column to langcode in url_alias.
+      // Rename the "language" column in the "url_alias" table to "langcode".
       if (db_table_exists('url_alias') && db_field_exists('url_alias', 'language')) {
+        // Drop the 2 indices that are based on the field we are renaming.
         db_drop_index('url_alias', 'alias_language_pid');
         db_drop_index('url_alias', 'source_language_pid');
+
         $langcode_spec = array(
           'description' => "The language code this alias is for; if 'und', the alias will be used for unknown languages. Each Backdrop path can have an alias for each supported language.",
           'type' => 'varchar',
           'length' => 12,
           'not null' => TRUE,
           'default' => '',
         );
-        $langcode_indexes = array('indexes' =>
-          array(
-            'alias_langcode_pid' => array('alias', 'langcode', 'pid'),
-            'source_langcode_pid' => array('source', 'langcode', 'pid'),
-          ),
-        );
-        db_change_field('url_alias', 'language', 'langcode', $langcode_spec, $langcode_indexes);
+        // Simply rename the "language" column to "langcode", without re-adding
+        // the indices we have removed earlier.
+        //
+        // Althought we could be doing all that simultaneously here (using the
+        // optional $keys_new parameter in the db_change_field() call), we are
+        // intentionally not doing it here, since that seems to be causing
+        // issues with certain outdated versions of MariaDB, which usually
+        // results in a broken 'url_alias' table (with an incorrect key file).
+        // See https://github.com/backdrop/backdrop-issues/issues/5496.
+        db_change_field('url_alias', 'language', 'langcode', $langcode_spec);
+
+        // Re-add the 2 indices separately here, after the "language" column has
+        // been renamed.
```

Comment:
I think the comments here can be simplified:

```suggestion
        // Re-add the 2 indices. We do this separately to avoid issues with
        // older versions of MariaDB.
        // @see https://github.com/backdrop/backdrop-issues/issues/5496
```